AlphaWallet/UI/ButtonsBar.swift

```diff
@@ -3,123 +3,201 @@
+            return []
+        }
+    } 
+    
+    var showMoreButton: Bool {
+        switch self {
+        case .green, .white:
+            return false
+        case .custom:
+            return false
+        case .combined(let buttons):
+            return buttons >= ButtonsBarConfiguration.maxCombinedButtons
+        case .empty:
+            return false
+        }
+    }
+}
+
+enum ButtonsBarButtonType {
+    case green
+    case white
+    case more
+    
+    var viewModel: ButtonsBarViewModel {
+        switch self {
+        case .green:
+            return ButtonsBarViewModel.greenButton
+        case .white:
+            return ButtonsBarViewModel.whiteButton
+        case .more:
+            return ButtonsBarViewModel.moreButton
+        }
+    }
+}
+
 class ButtonsBar: UIView {
     static let buttonsHeight = CGFloat(48)
     //A gap so it doesn't stick to the bottom of devices without a bottom safe area
     static let marginAtBottomScreen = CGFloat(3)
 
-    private var buttonContainerViews: [ContainerViewWithShadow<UIButton>]
+    private var buttonContainerViews: [ContainerViewWithShadow<UIButton>] = []
+    private var optionsButtonContainerViews: [ContainerViewWithShadow<UIButton>] = []
     private let buttonsStackView: UIStackView
 
-    var numberOfButtons: Int {
-        didSet {
-            buttonContainerViews = ButtonsBar.bar(numberOfButtons: numberOfButtons)
-            for each in buttonsStackView.arrangedSubviews {
-                each.removeFromSuperview()
-            }
-            buttonsStackView.addArrangedSubviews(buttons)
-        }
-    }
-
     var buttons: [UIButton] {
         return buttonContainerViews.map { $0.childView }
     }
+    
+    var optionButtons: [UIButton] {
+        return optionsButtonContainerViews.map { $0.childView }
+    }
 
     var isEmpty: Bool {
-        return numberOfButtons == 0
+        return configuration.buttonViewModels.isEmpty
     }
-
-    init(numberOfButtons: Int, buttonsDistribution: UIStackView.Distribution = .fillEqually) {
-        self.numberOfButtons = numberOfButtons
-        buttonsStackView =  [UIView]().asStackView(axis: .horizontal, distribution: buttonsDistribution, spacing: 7)
-        buttonContainerViews = ButtonsBar.bar(numberOfButtons: numberOfButtons)
-
+    
+    var configuration: ButtonsBarConfiguration = .empty {
+        didSet {
+            buttonContainerViews = ButtonsBar.bar(numberOfButtons: configuration.buttonViewModels.count)
+            optionsButtonContainerViews = ButtonsBar.bar(numberOfButtons: configuration.showMoreButton ? 1 : 0)
+            
+            for each in buttonsStackView.arrangedSubviews + innerStackView.arrangedSubviews {
+                each.removeFromSuperview()
+            }
+            
+            buttonsStackView.addArrangedSubviews(buttons)
+            innerStackView.addArrangedSubviews([buttonsStackView] + optionButtons)
+        }
+    }
+    
+    private var innerStackView: UIStackView
```

Comment:
- [ ] Can you move the private let/var upwards? So it's in this order. Just a bit easier to find:

private static

static

private var/let

private var/let (computed)

var/let

var/let (computed)